docs/source/conf.py

```diff
@@ -388,6 +389,47 @@ def _get_source_relative_path(source_abs_path):
     return os.path.relpath(source_abs_path, _find_source_root(source_abs_path))
 
 
+def _is_docstring_autosummary_compliant(docstring):
+    doc = docstring.split('\n')
+
+    # Extract until the first blank line if any.
+    try:
+        doc = doc[:doc.index('')]
```

Comment:
Is this correct?

Reply:
If `doc` does not have `''`, `index()` will raise `ValueError` (thus `doc` is not modified).
If it does, `index()` returns the index of `''`, so this statement extracts elements up until `''`, if any.

```
>>> a = ['a', 'b', 'c']                                                                                                                                                                                            
>>> a[:a.index('b')]                                                                                                                                                                                               
['a']
```

Reply:
OK, I understand.